lib/child_process.js

```diff
@@ -152,18 +150,17 @@ exports.exec = function exec(command /* , options, callback */) {
 
 const customPromiseExecFunction = (orig) => {
   return (...args) => {
-    const promise = createPromise();
-
-    orig(...args, (err, stdout, stderr) => {
-      if (err !== null) {
-        err.stdout = stdout;
-        err.stderr = stderr;
-        promiseReject(promise, err);
-      } else {
-        promiseResolve(promise, { stdout, stderr });
-      }
+    return new Promise((resolve, reject) => {
```

Comment:
Yes, we should measure the performance here.

I was suggesting to start with JS promises instead, since we spend some time recently to even inline the promise constructor into TurboFan optimized code. If that turns out to be too slow, then the other alternative is to use the createPromise and friends from `v8-extras`, which is also known to TurboFan, and significantly faster than going through C++.

Reply:
+1 for createPromise from v8-extras.